tempo_embeddings/text/corpus.py

```diff
@@ -65,6 +65,19 @@ def embeddings(self, embeddings: ArrayLike):
         self._embeddings = embeddings
         self._validate_embeddings()
 
+
+    def batches(self, batch_size: int) -> Iterable[list[Passage]]:
+        if batch_size <= 1:
+            yield self.passages
+        else:
+            for batch_start in tqdm(
+                range(0, len(self.passages), batch_size),
+                desc="Embeddings",
+                unit="batch",
+                total=len(self.passages) // batch_size + 1,
+            ):
+                yield self.passages[batch_start : batch_start + batch_size]
+
```

Comment:
I think the if/else is not needed (to avoid one level of nesting).

```suggestion
        if batch_size == 0:
            batch_size = len(self)

        for batch_start in tqdm(
            range(0, len(self.passages), batch_size),
            desc="Embeddings",
            unit="batch",
            total=len(self.passages) // batch_size + 1,
        ):
            yield self.passages[batch_start : batch_start + batch_size]
```